app/src/main/kotlin/com/wire/android/ui/userprofile/image/AvatarPicker.kt

```diff
@@ -37,18 +36,18 @@ import com.wire.android.ui.common.textfield.WirePrimaryButton
 import com.wire.android.ui.common.topappbar.WireCenterAlignedTopAppBar
 import com.wire.android.ui.theme.wireColorScheme
 import com.wire.android.ui.userprofile.image.AvatarPickerViewModel.ErrorCodes
-import com.wire.android.util.getWritableTempAvatarUri
 
 @OptIn(ExperimentalMaterialApi::class, ExperimentalMaterial3Api::class)
 @Composable
 fun AvatarPickerScreen(viewModel: AvatarPickerViewModel) {
-    val context = LocalContext.current
     val state = rememberAvatarPickerState()
```

Comment:
how about capturing a lambda here and instead of holding the PictureState in two sources, like it is right now, we just hold it inside the ViewModel, first we fetch the current avatar and change the status to the initial and later when this lambda is triggered we change the status to the PictureState.Picked(), that way we kind of have a single source for holding the state of the PictureState

Reply:
Done! Still have to polish some details, but so far, so good with the new approach 👍